### PR TITLE
add more types and docs!

### PR DESCRIPTION
**Pull request by mshwery**
This PR introduces more types and some documentation

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -4,13 +4,14 @@ declare module 'evergreen-ui' {
   import { IconName } from '@blueprintjs/icons'
   import * as React from 'react'
   import { extractStyles as boxExtractStyles } from 'ui-box'
-  import { BoxProps } from 'ui-box/dist/types/box-types'
+  import { BoxProps, Is } from 'ui-box/dist/types/box-types'
```
Comment (Rowno) [inline index.d.ts]: Maybe `ui-box` should properly export these types so you can just do:
```
import { BoxProps, Is } from 'ui-box'
```

File: index.d.ts [hunk 2]
```diff
@@ -120,12 +522,37 @@ declare module 'evergreen-ui' {
   export class Card extends React.PureComponent<CardProps> {
   }
 
+  export interface CodeProps extends TextProps<'code'> {
+
+  }
```
Comment (Rowno) [inline index.d.ts]: Might be nicer to use `type CodeProps = TextProps<'code'>`?

File: index.d.ts (hunk 1 above)
Comment (mshwery) [reply index.d.ts]: Yes! Would be better.

File: index.d.ts (hunk 2 above)
Comment (mshwery) [reply index.d.ts]: Ok!

**Pull request merged**